lib/models/application.coffee

```diff
@@ -694,6 +694,128 @@ getApplicationModel = (deps, opts) ->
 			throw err
 		.asCallback(callback)
 
+	###*
+	# @summary Check if rolling release is enabled for specific application
+	# @name isRollingReleaseEnabled
+	# @public
+	# @function
+	# @memberof resin.models.application
+	#
+	# @param {String|Number} nameOrId - application name (string) or id (number)
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.application.isRollingReleaseEnabled('MyApp').then(function(isEnabled) {
+	# 	console.log(isEnabled);
+	# });
+	###
+
+	exports.isRollingReleaseEnabled = (nameOrId, callback) ->
+		exports.get(nameOrId, $select: 'should_track_latest_release')
+		.get('should_track_latest_release').asCallback(callback)
+
+	###*
+	# @summary Set application to particular commit
+	# @name setApplicationToRelease
+	# @public
+	# @function
+	# @memberof resin.models.application
+	#
+	# @param {String|Number} nameOrId - application name (string) or id (number)
+	#
+	# @description this method requires rolling release to be disabled prior to call
+	#
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.application.disableRollingRelease('MyApp').then(
+	# 	resin.models.application.isRollingReleaseEnabled('MyApp').then(function(isEnabled) {
+	# 		console.log(isEnabled);
+	# 	})
+	# );
+	###
+
+	exports.setApplicationToRelease = (nameOrId, fullCommitHash, callback) ->
+		getId(nameOrId)
+		.then (applicationId) ->
+			exports.isRollingReleaseEnabled(applicationId)
+			.then (isEnabled) ->
+				if isEnabled
+					throw new Error ('Rolling release enabled')
+
+				return pine.patch
+					resource: 'application'
+					id: applicationId
+					body: commit: fullCommitHash
+	  .asCallback(callback)
+
+	###*
+	# @summary Get current release for specific application
+	# @name getRelease
+	# @public
+	# @function
+	# @memberof resin.models.application
+	#
+	# @param {String|Number} nameOrId - application name (string) or id (number)
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.application.getRelease('MyApp').then(function(release) {
+	# 	console.log(release);
+	# });
+	###
+
+	exports.getRelease = (nameOrId, callback) ->
+		exports.get(nameOrId)
```

Comment:
If we only want the `commit` field, it'd be good to `$select: 'commit'` here. It works anyway, but that way we avoid extra work for the API & DB, and avoid transferring lots of unnecessary data.